upgrade-notes/backend_20240722_151150.md

```diff
@@ -0,0 +1,20 @@
+#### add customer user login types to the data layer ([#3276](https://github.com/shopsys/shopsys/pull/3276))
+
+-   `Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser` class was changed:
+    -   `$lastLogin` property and `getLastLogin()` method were removed, use `CustomerUserLoginInformationProvider::getLastLogin()` instead
+    -   `onLogin()` method was removed
+-   `Shopsys\FrontendApiBundle\Model\Mutation\Customer\User\CustomerUserMutation::__construct()` method changed its interface:
+    ```diff
+        public function __construct(
+            // ...
+    +       protected readonly CustomerUserLoginTypeFacade $customerUserLoginTypeFacade,
+    +       protected readonly CustomerUserLoginTypeDataFactory $customerUserLoginTypeDataFactory,
+    ```
+-   `Shopsys\FrontendApiBundle\Model\Mutation\Login\LoginMutation::__construct()` method changed its interface:
+    ```diff
+        public function __construct(
+            // ...
+    +       protected readonly CustomerUserLoginTypeFacade $customerUserLoginTypeFacade,
+    +       protected readonly CustomerUserLoginTypeDataFactory $customerUserLoginTypeDataFactory,
+    ```
```

Comment:
You should use full diff not just partial as it is not clear in what place were these properties added.

Reply:
fixed in [1f9542f](https://github.com/shopsys/shopsys/pull/3276/commits/1f9542f3895cdf060d3480a0d841daa2352d11fa)